src/libstd/macros.rs

```diff
@@ -64,6 +64,9 @@ macro_rules! panic {
 ///
 /// Equivalent to the `println!` macro except that a newline is not printed at
 /// the end of the message.
+///
+/// Note that because stdout is by default buffered on unix systems, this may
+/// result in output not being emitted until stdout is flushed.
```

Comment:
Unfortunately, no, there is no default auto-linking, but you can always do it manually via `[foo](../bar/baz.html)`